From patchwork Wed Oct 10 22:36:48 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit
X-Patchwork-Submitter: =?utf-8?q?Philippe_Mathieu-Daud=C3=A9?=
 <f4bug@amsat.org>
X-Patchwork-Id: 10635401
Return-Path: 
 <qemu-devel-bounces+patchwork-qemu-devel=patchwork.kernel.org@nongnu.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 0AD4269B1
	for <patchwork-qemu-devel@patchwork.kernel.org>;
 Wed, 10 Oct 2018 22:41:34 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id EC5F02AC4A
	for <patchwork-qemu-devel@patchwork.kernel.org>;
 Wed, 10 Oct 2018 22:41:33 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id E09142AC5D; Wed, 10 Oct 2018 22:41:33 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.7 required=2.0 tests=BAYES_00,DKIM_INVALID,
	DKIM_SIGNED,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from lists.gnu.org (lists.gnu.org [208.118.235.17])
	(using TLSv1 with cipher AES256-SHA (256/256 bits))
	(No client certificate requested)
	by mail.wl.linuxfoundation.org (Postfix) with ESMTPS id 96FD62AC4A
	for <patchwork-qemu-devel@patchwork.kernel.org>;
 Wed, 10 Oct 2018 22:41:33 +0000 (UTC)
Received: from localhost ([::1]:59451 helo=lists.gnu.org)
	by lists.gnu.org with esmtp (Exim 4.71)
	(envelope-from
 <qemu-devel-bounces+patchwork-qemu-devel=patchwork.kernel.org@nongnu.org>)
	id 1gANAb-0005vl-08
	for patchwork-qemu-devel@patchwork.kernel.org;
 Wed, 10 Oct 2018 18:41:33 -0400
Received: from eggs.gnu.org ([2001:4830:134:3::10]:38757)
	by lists.gnu.org with esmtp (Exim 4.71)
	(envelope-from <philippe.mathieu.daude@gmail.com>)
	id 1gAN6K-0002dh-Li
	for qemu-devel@nongnu.org; Wed, 10 Oct 2018 18:37:09 -0400
Received: from Debian-exim by eggs.gnu.org with spam-scanned (Exim 4.71)
	(envelope-from <philippe.mathieu.daude@gmail.com>)
	id 1gAN6H-0000J9-TI
	for qemu-devel@nongnu.org; Wed, 10 Oct 2018 18:37:08 -0400
Received: from mail-wm1-x32e.google.com ([2a00:1450:4864:20::32e]:36302)
	by eggs.gnu.org with esmtps (TLS1.0:RSA_AES_128_CBC_SHA1:16)
	(Exim 4.71) (envelope-from <philippe.mathieu.daude@gmail.com>)
	id 1gAN6G-0000I4-KB
	for qemu-devel@nongnu.org; Wed, 10 Oct 2018 18:37:05 -0400
Received: by mail-wm1-x32e.google.com with SMTP id a8-v6so7318473wmf.1
	for <qemu-devel@nongnu.org>; Wed, 10 Oct 2018 15:37:04 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=gmail.com; s=20161025;
	h=sender:from:to:cc:subject:date:message-id:in-reply-to:references
	:mime-version:content-transfer-encoding;
	bh=YYHzCMBLWBqfoBZPbPvnMrKs+40SHNKrFAhfcyF3gjk=;
	b=UyXze8pf+zGc4f2VWKYHMO17KfvUdp7fpTCh1cVYEMUXOg4kakOofgiA0W8rUrcEXg
	6jKEhmtKhtR5x0tsVdhOfxqeBqW4GHc855LHTGYnHtPiMjxSQ3V4IR8F0onMpR9h3XTS
	Bj2TeS1PGaZG1xIve4fPeOIbvJ+G0CBQsYcaATcQ0Q6UuDjj8eEzFG+dNzx3ypamvcYU
	9BZJkT6kwQoopWpMC1I/VK2e6aEJZtUUCiuvh9KalPGNCHlUAI9+LtUXNB4SI6bPmW4f
	eVEH82mmbn7rEapOhCKlIhdr3yfQKBTypkyLyJOKhJ6mUzeBo2ki+z9PApSvlIKmIt2o
	13FQ==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
	d=1e100.net; s=20161025;
	h=x-gm-message-state:sender:from:to:cc:subject:date:message-id
	:in-reply-to:references:mime-version:content-transfer-encoding;
	bh=YYHzCMBLWBqfoBZPbPvnMrKs+40SHNKrFAhfcyF3gjk=;
	b=sERnceL13P5QjmPI274/Gjw17H/Nzh+GhhjL6oxX/2lqqV9PCwAwOBO84ApxMw097R
	9+1BOZFt+k2H1o2opSBAR8BgroDBrq7hHAjER/zWnqlKIw/mbLofSo5C6UT1pLfX2XGH
	YKYX8rusoeSC0KeDC2CCzxTyp1g9kxjFhNdvIVHXx0E4BWsQKbkjI2Lf+fxtx+1mpxPL
	7IAUsFyFcGWr3YpDf/V3FrQv+1OXceWm55M90EaYd1lh1E2UzucZGCwi4g1zk++ChhoI
	6cMo3RICZImwjvee2TBUtBI7iiIccHEnERNTy7lD4uCj0ouz/xkmN8iTbPi4tWwVrh7Z
	26gQ==
X-Gm-Message-State: ABuFfoguyLvIEOIL6bGJXB4RALOp/91lvvxNzJVUxt6e2fTw3KJ4gzxM
	sYQ/lZK42/n1lmQuCba6FX4=
X-Google-Smtp-Source: 
 ACcGV63V6wMvwfeJcd7jyNfzy+Z9EMXqqlOwCfSncJHYvPnCpNN7QGfOhaBivZJ4BNIKZLnxrVKkIA==
X-Received: by 2002:a1c:6783:: with SMTP id
	b125-v6mr2425243wmc.142.1539211023108;
	Wed, 10 Oct 2018 15:37:03 -0700 (PDT)
Received: from x1.local (26.red-83-32-208.dynamicip.rima-tde.net.
	[83.32.208.26]) by smtp.gmail.com with ESMTPSA id
	q77-v6sm10344886wmd.33.2018.10.10.15.37.02
	(version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
	Wed, 10 Oct 2018 15:37:02 -0700 (PDT)
From: =?utf-8?q?Philippe_Mathieu-Daud=C3=A9?= <f4bug@amsat.org>
To: Laurent Vivier <laurent@vivier.eu>
Date: Thu, 11 Oct 2018 00:36:48 +0200
Message-Id: <20181010223656.31632-3-f4bug@amsat.org>
X-Mailer: git-send-email 2.19.1
In-Reply-To: <20181010223656.31632-1-f4bug@amsat.org>
References: <20181010223656.31632-1-f4bug@amsat.org>
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
X-detected-operating-system: by eggs.gnu.org: Genre and OS details not
	recognized.
X-Received-From: 2a00:1450:4864:20::32e
Subject: [Qemu-devel] [PATCH v5 02/10] linux-user/strace: Display invalid
 pointer in print_timeval()
X-BeenThere: qemu-devel@nongnu.org
X-Mailman-Version: 2.1.21
Precedence: list
List-Id: <qemu-devel.nongnu.org>
List-Unsubscribe: <https://lists.nongnu.org/mailman/options/qemu-devel>,
	<mailto:qemu-devel-request@nongnu.org?subject=unsubscribe>
List-Archive: <http://lists.nongnu.org/archive/html/qemu-devel/>
List-Post: <mailto:qemu-devel@nongnu.org>
List-Help: <mailto:qemu-devel-request@nongnu.org?subject=help>
List-Subscribe: <https://lists.nongnu.org/mailman/listinfo/qemu-devel>,
	<mailto:qemu-devel-request@nongnu.org?subject=subscribe>
Cc: =?utf-8?q?Guido_G=C3=BCnther?= <agx@sigxcpu.org>,
 Riku Voipio <riku.voipio@iki.fi>,
 =?utf-8?q?Philippe_Mathieu-Daud=C3=A9?= <f4bug@amsat.org>,
 qemu-devel@nongnu.org
Errors-To: 
 qemu-devel-bounces+patchwork-qemu-devel=patchwork.kernel.org@nongnu.org
Sender: "Qemu-devel"
	<qemu-devel-bounces+patchwork-qemu-devel=patchwork.kernel.org@nongnu.org>
X-Virus-Scanned: ClamAV using ClamSMTP

Suggested-by: Laurent Vivier <laurent@vivier.eu>
Signed-off-by: Philippe Mathieu-Daudé <f4bug@amsat.org>
---
 linux-user/strace.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/linux-user/strace.c b/linux-user/strace.c
index 33f4a506a2..c95557bf83 100644
--- a/linux-user/strace.c
+++ b/linux-user/strace.c
@@ -1173,8 +1173,10 @@ print_timeval(abi_ulong tv_addr, int last)
         struct target_timeval *tv;
 
         tv = lock_user(VERIFY_READ, tv_addr, sizeof(*tv), 1);
-        if (!tv)
+        if (!tv) {
+            print_pointer(tv_addr, last);
             return;
+        }
         gemu_log("{" TARGET_ABI_FMT_ld "," TARGET_ABI_FMT_ld "}%s",
             tswapal(tv->tv_sec), tswapal(tv->tv_usec), get_comma(last));
         unlock_user(tv, tv_addr, 0);

From patchwork Wed Oct 10 22:36:49 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit
X-Patchwork-Submitter: =?utf-8?q?Philippe_Mathieu-Daud=C3=A9?=
 <f4bug@amsat.org>
X-Patchwork-Id: 10635391
Return-Path: 
 <qemu-devel-bounces+patchwork-qemu-devel=patchwork.kernel.org@nongnu.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id BD04815E2
	for <patchwork-qemu-devel@patchwork.kernel.org>;
 Wed, 10 Oct 2018 22:38:46 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id A89FD2AC4A
	for <patchwork-qemu-devel@patchwork.kernel.org>;
 Wed, 10 Oct 2018 22:38:46 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 9C1DC2AC54; Wed, 10 Oct 2018 22:38:46 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.7 required=2.0 tests=BAYES_00,DKIM_INVALID,
	DKIM_SIGNED,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from lists.gnu.org (lists.gnu.org [208.118.235.17])
	(using TLSv1 with cipher AES256-SHA (256/256 bits))
	(No client certificate requested)
	by mail.wl.linuxfoundation.org (Postfix) with ESMTPS id 316362AC46
	for <patchwork-qemu-devel@patchwork.kernel.org>;
 Wed, 10 Oct 2018 22:38:46 +0000 (UTC)
Received: from localhost ([::1]:59433 helo=lists.gnu.org)
	by lists.gnu.org with esmtp (Exim 4.71)
	(envelope-from
 <qemu-devel-bounces+patchwork-qemu-devel=patchwork.kernel.org@nongnu.org>)
	id 1gAN7s-0003nh-T6
	for patchwork-qemu-devel@patchwork.kernel.org;
 Wed, 10 Oct 2018 18:38:44 -0400
Received: from eggs.gnu.org ([2001:4830:134:3::10]:38754)
	by lists.gnu.org with esmtp (Exim 4.71)
	(envelope-from <philippe.mathieu.daude@gmail.com>)
	id 1gAN6K-0002df-LS
	for qemu-devel@nongnu.org; Wed, 10 Oct 2018 18:37:09 -0400
Received: from Debian-exim by eggs.gnu.org with spam-scanned (Exim 4.71)
	(envelope-from <philippe.mathieu.daude@gmail.com>)
	id 1gAN6I-0000JX-Hx
	for qemu-devel@nongnu.org; Wed, 10 Oct 2018 18:37:08 -0400
Received: from mail-wr1-x42c.google.com ([2a00:1450:4864:20::42c]:42847)
	by eggs.gnu.org with esmtps (TLS1.0:RSA_AES_128_CBC_SHA1:16)
	(Exim 4.71) (envelope-from <philippe.mathieu.daude@gmail.com>)
	id 1gAN6H-0000Id-SS
	for qemu-devel@nongnu.org; Wed, 10 Oct 2018 18:37:06 -0400
Received: by mail-wr1-x42c.google.com with SMTP id g15-v6so7444690wru.9
	for <qemu-devel@nongnu.org>; Wed, 10 Oct 2018 15:37:05 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=gmail.com; s=20161025;
	h=sender:from:to:cc:subject:date:message-id:in-reply-to:references
	:mime-version:content-transfer-encoding;
	bh=ogKL/TXLX/jLCwe+X43K+U+zYiSlou+fzYLs4QPUzns=;
	b=qD3MDspP3MpofjJoUyL2ixPS4rd3X9JAlOfhHzeAKToTQIH6iFeNFhLMZbib27os6n
	5rqrYoz/XxY5GVkigUKmGBzwe7YR8R6M1eHPZzmAAoopvfsm9zddCgLiN+qNu7zIoOEC
	GpwnWyC877epwCgHAKyAoYSK1OWMNQVL8g5SMoLjcnzTRuM1AuKP1HA6H//HT6dxrpN4
	HPtmIhHnN6kvYRRiZsNgHLr3bx5Wtn05fvRmYIO90T/jZF4jChCjpyGuXyhI6+gztVY+
	dMcbC8xPLHAR6vIe5YCtN1RO/lo/wfs9FIaj2fvem1SF0gWMyaXrP7AIYfYZm6q/QUXL
	JNcA==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
	d=1e100.net; s=20161025;
	h=x-gm-message-state:sender:from:to:cc:subject:date:message-id
	:in-reply-to:references:mime-version:content-transfer-encoding;
	bh=ogKL/TXLX/jLCwe+X43K+U+zYiSlou+fzYLs4QPUzns=;
	b=I/VvfNrLNo/R77WTW4QKu0PSbKgtF8d/qjQlNOr6IMMDWv7e92DytY9+bHs1Gm2HBb
	l/Ry5EeBxjUgLGykGcZZVZtsXMeaOTTzW/8UE7ADIJi6luFtQWUI59kBWGcpjer7v4Ez
	LAj/O9NFIq7k9oLdV5h/L8to3LxuDigBgFHs37waAv2bZcMUvL5FFcvUqcVfgiDKYryX
	1QEZxPVVQ3Xhl3hVtK0C4xijpHPcvQgdlRfmmzFeTpxGPqMKHwNOZoQCSq6tQjdevHgU
	+3J7KlKfudaS1dg1SJ47rXmsVG4fS38OmdKB6mDFGNYsHG5TyV6ZQ5hnIdwqWN9a3GMG
	h6VQ==
X-Gm-Message-State: ABuFfojiVzTWjKRd82t7G7sO1r6dv6Mq3rF2eo0HdCbDsVdvvXtYnPzJ
	k9wMk6dAqGQSxL+nW0H5aJk=
X-Google-Smtp-Source: 
 ACcGV60VQpjrxXHksT3VFf+DPj/vq5kiSMj7XKGSPd8YQ/49/pvlDxYwTtJhOwN+zB/PbXFSQv13mw==
X-Received: by 2002:adf:90ee:: with SMTP id
	i101-v6mr24535174wri.181.1539211024309;
	Wed, 10 Oct 2018 15:37:04 -0700 (PDT)
Received: from x1.local (26.red-83-32-208.dynamicip.rima-tde.net.
	[83.32.208.26]) by smtp.gmail.com with ESMTPSA id
	q77-v6sm10344886wmd.33.2018.10.10.15.37.03
	(version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
	Wed, 10 Oct 2018 15:37:03 -0700 (PDT)
From: =?utf-8?q?Philippe_Mathieu-Daud=C3=A9?= <f4bug@amsat.org>
To: Laurent Vivier <laurent@vivier.eu>
Date: Thu, 11 Oct 2018 00:36:49 +0200
Message-Id: <20181010223656.31632-4-f4bug@amsat.org>
X-Mailer: git-send-email 2.19.1
In-Reply-To: <20181010223656.31632-1-f4bug@amsat.org>
References: <20181010223656.31632-1-f4bug@amsat.org>
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
X-detected-operating-system: by eggs.gnu.org: Genre and OS details not
	recognized.
X-Received-From: 2a00:1450:4864:20::42c
Subject: [Qemu-devel] [PATCH v5 03/10] linux-user/strace: Add
 print_timezone()
X-BeenThere: qemu-devel@nongnu.org
X-Mailman-Version: 2.1.21
Precedence: list
List-Id: <qemu-devel.nongnu.org>
List-Unsubscribe: <https://lists.nongnu.org/mailman/options/qemu-devel>,
	<mailto:qemu-devel-request@nongnu.org?subject=unsubscribe>
List-Archive: <http://lists.nongnu.org/archive/html/qemu-devel/>
List-Post: <mailto:qemu-devel@nongnu.org>
List-Help: <mailto:qemu-devel-request@nongnu.org?subject=help>
List-Subscribe: <https://lists.nongnu.org/mailman/listinfo/qemu-devel>,
	<mailto:qemu-devel-request@nongnu.org?subject=subscribe>
Cc: =?utf-8?q?Guido_G=C3=BCnther?= <agx@sigxcpu.org>,
 Riku Voipio <riku.voipio@iki.fi>,
 =?utf-8?q?Philippe_Mathieu-Daud=C3=A9?= <f4bug@amsat.org>,
 qemu-devel@nongnu.org
Errors-To: 
 qemu-devel-bounces+patchwork-qemu-devel=patchwork.kernel.org@nongnu.org
Sender: "Qemu-devel"
	<qemu-devel-bounces+patchwork-qemu-devel=patchwork.kernel.org@nongnu.org>
X-Virus-Scanned: ClamAV using ClamSMTP

Suggested-by: Laurent Vivier <laurent@vivier.eu>
Signed-off-by: Philippe Mathieu-Daudé <f4bug@amsat.org>
---
 linux-user/strace.c | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/linux-user/strace.c b/linux-user/strace.c
index c95557bf83..10ae22263c 100644
--- a/linux-user/strace.c
+++ b/linux-user/strace.c
@@ -63,6 +63,7 @@ UNUSED static void print_string(abi_long, int);
 UNUSED static void print_buf(abi_long addr, abi_long len, int last);
 UNUSED static void print_raw_param(const char *, abi_long, int);
 UNUSED static void print_timeval(abi_ulong, int);
+UNUSED static void print_timezone(abi_ulong, int);
 UNUSED static void print_number(abi_long, int);
 UNUSED static void print_signal(abi_ulong, int);
 UNUSED static void print_sockaddr(abi_ulong addr, abi_long addrlen);
@@ -1184,6 +1185,26 @@ print_timeval(abi_ulong tv_addr, int last)
         gemu_log("NULL%s", get_comma(last));
 }
 
+static void
+print_timezone(abi_ulong tz_addr, int last)
+{
+    if (tz_addr) {
+        struct target_timezone *tz;
+
+        tz = lock_user(VERIFY_READ, tz_addr, sizeof(*tz), 1);
+        if (!tz) {
+            print_pointer(tz_addr, last);
+            return;
+        }
+        gemu_log("{" TARGET_ABI_FMT_ld "," TARGET_ABI_FMT_ld "}%s",
+                 tswapal(tz->tz_minuteswest), tswapal(tz->tz_dsttime),
+                 get_comma(last));
+        unlock_user(tz, tz_addr, 0);
+    } else {
+        gemu_log("NULL%s", get_comma(last));
+    }
+}
+
 #undef UNUSED
 
 #ifdef TARGET_NR_accept

From patchwork Wed Oct 10 22:36:50 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit
X-Patchwork-Submitter: =?utf-8?q?Philippe_Mathieu-Daud=C3=A9?=
 <f4bug@amsat.org>
X-Patchwork-Id: 10635403
Return-Path: 
 <qemu-devel-bounces+patchwork-qemu-devel=patchwork.kernel.org@nongnu.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 496B8112B
	for <patchwork-qemu-devel@patchwork.kernel.org>;
 Wed, 10 Oct 2018 22:41:36 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 2CED62AC4A
	for <patchwork-qemu-devel@patchwork.kernel.org>;
 Wed, 10 Oct 2018 22:41:36 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 20F8E2AC5D; Wed, 10 Oct 2018 22:41:36 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.7 required=2.0 tests=BAYES_00,DKIM_INVALID,
	DKIM_SIGNED,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from lists.gnu.org (lists.gnu.org [208.118.235.17])
	(using TLSv1 with cipher AES256-SHA (256/256 bits))
	(No client certificate requested)
	by mail.wl.linuxfoundation.org (Postfix) with ESMTPS id C120D2AC4A
	for <patchwork-qemu-devel@patchwork.kernel.org>;
 Wed, 10 Oct 2018 22:41:35 +0000 (UTC)
Received: from localhost ([::1]:59452 helo=lists.gnu.org)
	by lists.gnu.org with esmtp (Exim 4.71)
	(envelope-from
 <qemu-devel-bounces+patchwork-qemu-devel=patchwork.kernel.org@nongnu.org>)
	id 1gANAd-0005wp-3M
	for patchwork-qemu-devel@patchwork.kernel.org;
 Wed, 10 Oct 2018 18:41:35 -0400
Received: from eggs.gnu.org ([2001:4830:134:3::10]:38791)
	by lists.gnu.org with esmtp (Exim 4.71)
	(envelope-from <philippe.mathieu.daude@gmail.com>)
	id 1gAN6N-0002ee-9g
	for qemu-devel@nongnu.org; Wed, 10 Oct 2018 18:37:12 -0400
Received: from Debian-exim by eggs.gnu.org with spam-scanned (Exim 4.71)
	(envelope-from <philippe.mathieu.daude@gmail.com>)
	id 1gAN6K-0000N5-LM
	for qemu-devel@nongnu.org; Wed, 10 Oct 2018 18:37:10 -0400
Received: from mail-wm1-x335.google.com ([2a00:1450:4864:20::335]:55215)
	by eggs.gnu.org with esmtps (TLS1.0:RSA_AES_128_CBC_SHA1:16)
	(Exim 4.71) (envelope-from <philippe.mathieu.daude@gmail.com>)
	id 1gAN6J-0000JD-Eg
	for qemu-devel@nongnu.org; Wed, 10 Oct 2018 18:37:08 -0400
Received: by mail-wm1-x335.google.com with SMTP id r63-v6so7064151wma.4
	for <qemu-devel@nongnu.org>; Wed, 10 Oct 2018 15:37:06 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=gmail.com; s=20161025;
	h=sender:from:to:cc:subject:date:message-id:in-reply-to:references
	:mime-version:content-transfer-encoding;
	bh=3kAOsYRiFVJksmPZLwxebdoJBKsX2VAhyUpwsb1TMRI=;
	b=PlIqcG1DTMWHm8Tt8P87+ALdWwalnHqIjGpd5NRSN0DzLlBCZOafuBc6z9Pam8daqC
	momtECl/MLgJYDp/p+II0kEApyxWZ21V7Rq2it+qI74YNUTTDqkjwWtgmPl2SXScDhfM
	DXWYNf+Yfw1wblQpdNjKhAIa1ESLJeSpmV8SC+7fFLCzGXKjRfFGgP+gPLM+1EFN7LcX
	wN/5Pauw4+W+LqXYqtaglvrb4f3PKTdCI9mtj15SXZCyyxuyY96uJUL7tECUBJFEvCMO
	8nqcR+fAHtkPt/NllkE7RBB9lSQBZ7ZH+RQ3wFWv+78xa74eTR0gApdU90spJhLPvbsG
	EcRQ==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
	d=1e100.net; s=20161025;
	h=x-gm-message-state:sender:from:to:cc:subject:date:message-id
	:in-reply-to:references:mime-version:content-transfer-encoding;
	bh=3kAOsYRiFVJksmPZLwxebdoJBKsX2VAhyUpwsb1TMRI=;
	b=dVVoQaEwrbkj7aCyaWg/kxXICLaly/Vy7bqwMmT3Pf/dXvB1s00jufgzKZrM7Z+f/t
	RjowbblnBtKXiD4/hC3CwYUC4yxZrlFNDger75aiAZ/WHtaguM43Z4Yl3nbRDCMQYsSS
	YY3NSQTUnCn1RTk4xZSyZuDuTW2WE8otgDL+Ag5utSgAXoVliAM4fKkr6YrEpW1CPHQu
	Qo0LDmSzqm8+wiuwLyxSq7BfoxkZd2rFHe5sHraUe+ntL+eREFCjAkBnFzfxT9VTXlvL
	Zd7OReOO9hwlp5hIq3BPTGL58SNswHCsdIlaxijj3m6Ka2glU5ibkcGdEf+MXEws+NTa
	klCA==
X-Gm-Message-State: ABuFfohIerNqz7FbQAwlFHHhKTXbfrETaoMqt5cDR4pXp4BA54ohqrUU
	PC1KuPPa2UtKHYWlVdSEx3Q=
X-Google-Smtp-Source: 
 ACcGV63b+o2LAyqdXePslfHY3MNf7C963lpPnWyPVqyjb0YN+Dd/3fNbHXcvEyq7ijwLMIUzBXDYig==
X-Received: by 2002:a1c:9901:: with SMTP id
 b1-v6mr2246629wme.15.1539211025590;
	Wed, 10 Oct 2018 15:37:05 -0700 (PDT)
Received: from x1.local (26.red-83-32-208.dynamicip.rima-tde.net.
	[83.32.208.26]) by smtp.gmail.com with ESMTPSA id
	q77-v6sm10344886wmd.33.2018.10.10.15.37.04
	(version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
	Wed, 10 Oct 2018 15:37:04 -0700 (PDT)
From: =?utf-8?q?Philippe_Mathieu-Daud=C3=A9?= <f4bug@amsat.org>
To: Laurent Vivier <laurent@vivier.eu>
Date: Thu, 11 Oct 2018 00:36:50 +0200
Message-Id: <20181010223656.31632-5-f4bug@amsat.org>
X-Mailer: git-send-email 2.19.1
In-Reply-To: <20181010223656.31632-1-f4bug@amsat.org>
References: <20181010223656.31632-1-f4bug@amsat.org>
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
X-detected-operating-system: by eggs.gnu.org: Genre and OS details not
	recognized.
X-Received-From: 2a00:1450:4864:20::335
Subject: [Qemu-devel] [PATCH v5 04/10] linux-user/strace: Improve
 settimeofday()
X-BeenThere: qemu-devel@nongnu.org
X-Mailman-Version: 2.1.21
Precedence: list
List-Id: <qemu-devel.nongnu.org>
List-Unsubscribe: <https://lists.nongnu.org/mailman/options/qemu-devel>,
	<mailto:qemu-devel-request@nongnu.org?subject=unsubscribe>
List-Archive: <http://lists.nongnu.org/archive/html/qemu-devel/>
List-Post: <mailto:qemu-devel@nongnu.org>
List-Help: <mailto:qemu-devel-request@nongnu.org?subject=help>
List-Subscribe: <https://lists.nongnu.org/mailman/listinfo/qemu-devel>,
	<mailto:qemu-devel-request@nongnu.org?subject=subscribe>
Cc: =?utf-8?q?Guido_G=C3=BCnther?= <agx@sigxcpu.org>,
 Riku Voipio <riku.voipio@iki.fi>,
 =?utf-8?q?Philippe_Mathieu-Daud=C3=A9?= <f4bug@amsat.org>,
 qemu-devel@nongnu.org
Errors-To: 
 qemu-devel-bounces+patchwork-qemu-devel=patchwork.kernel.org@nongnu.org
Sender: "Qemu-devel"
	<qemu-devel-bounces+patchwork-qemu-devel=patchwork.kernel.org@nongnu.org>
X-Virus-Scanned: ClamAV using ClamSMTP

Signed-off-by: Philippe Mathieu-Daudé <f4bug@amsat.org>
Tested-By: Guido Günther <agx@sigxcpu.org>
Reviewed-by: Laurent Vivier <laurent@vivier.eu>
---
 linux-user/strace.c    | 13 +++++++++++++
 linux-user/strace.list |  2 +-
 2 files changed, 14 insertions(+), 1 deletion(-)

diff --git a/linux-user/strace.c b/linux-user/strace.c
index 10ae22263c..ca2cd75993 100644
--- a/linux-user/strace.c
+++ b/linux-user/strace.c
@@ -1502,6 +1502,19 @@ print_futimesat(const struct syscallname *name,
 }
 #endif
 
+#ifdef TARGET_NR_settimeofday
+static void
+print_settimeofday(const struct syscallname *name,
+                abi_long arg0, abi_long arg1, abi_long arg2,
+                abi_long arg3, abi_long arg4, abi_long arg5)
+{
+    print_syscall_prologue(name);
+    print_timeval(arg0, 0);
+    print_timezone(arg1, 1);
+    print_syscall_epilogue(name);
+}
+#endif
+
 #ifdef TARGET_NR_link
 static void
 print_link(const struct syscallname *name,
diff --git a/linux-user/strace.list b/linux-user/strace.list
index ff8bb19f5f..31093c3371 100644
--- a/linux-user/strace.list
+++ b/linux-user/strace.list
@@ -1345,7 +1345,7 @@
 { TARGET_NR_set_tid_address, "set_tid_address" , NULL, NULL, NULL },
 #endif
 #ifdef TARGET_NR_settimeofday
-{ TARGET_NR_settimeofday, "settimeofday" , NULL, NULL, NULL },
+{ TARGET_NR_settimeofday, "settimeofday" , NULL, print_settimeofday, NULL },
 #endif
 #ifdef TARGET_NR_setuid
 { TARGET_NR_setuid, "setuid" , NULL, NULL, NULL },

From patchwork Wed Oct 10 22:36:51 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit
X-Patchwork-Submitter: =?utf-8?q?Philippe_Mathieu-Daud=C3=A9?=
 <f4bug@amsat.org>
X-Patchwork-Id: 10635399
Return-Path: 
 <qemu-devel-bounces+patchwork-qemu-devel=patchwork.kernel.org@nongnu.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 87E39112B
	for <patchwork-qemu-devel@patchwork.kernel.org>;
 Wed, 10 Oct 2018 22:41:33 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 6E5D32AC4A
	for <patchwork-qemu-devel@patchwork.kernel.org>;
 Wed, 10 Oct 2018 22:41:33 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 5F3F32AC5D; Wed, 10 Oct 2018 22:41:33 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.7 required=2.0 tests=BAYES_00,DKIM_INVALID,
	DKIM_SIGNED,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from lists.gnu.org (lists.gnu.org [208.118.235.17])
	(using TLSv1 with cipher AES256-SHA (256/256 bits))
	(No client certificate requested)
	by mail.wl.linuxfoundation.org (Postfix) with ESMTPS id 09B7C2AC4A
	for <patchwork-qemu-devel@patchwork.kernel.org>;
 Wed, 10 Oct 2018 22:41:33 +0000 (UTC)
Received: from localhost ([::1]:59450 helo=lists.gnu.org)
	by lists.gnu.org with esmtp (Exim 4.71)
	(envelope-from
 <qemu-devel-bounces+patchwork-qemu-devel=patchwork.kernel.org@nongnu.org>)
	id 1gANAa-0005vK-Bj
	for patchwork-qemu-devel@patchwork.kernel.org;
 Wed, 10 Oct 2018 18:41:32 -0400
Received: from eggs.gnu.org ([2001:4830:134:3::10]:38793)
	by lists.gnu.org with esmtp (Exim 4.71)
	(envelope-from <philippe.mathieu.daude@gmail.com>)
	id 1gAN6N-0002eh-9o
	for qemu-devel@nongnu.org; Wed, 10 Oct 2018 18:37:12 -0400
Received: from Debian-exim by eggs.gnu.org with spam-scanned (Exim 4.71)
	(envelope-from <philippe.mathieu.daude@gmail.com>)
	id 1gAN6K-0000NU-RL
	for qemu-devel@nongnu.org; Wed, 10 Oct 2018 18:37:10 -0400
Received: from mail-wm1-x332.google.com ([2a00:1450:4864:20::332]:37052)
	by eggs.gnu.org with esmtps (TLS1.0:RSA_AES_128_CBC_SHA1:16)
	(Exim 4.71) (envelope-from <philippe.mathieu.daude@gmail.com>)
	id 1gAN6K-0000Jk-Ha
	for qemu-devel@nongnu.org; Wed, 10 Oct 2018 18:37:08 -0400
Received: by mail-wm1-x332.google.com with SMTP id 185-v6so7375558wmt.2
	for <qemu-devel@nongnu.org>; Wed, 10 Oct 2018 15:37:08 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=gmail.com; s=20161025;
	h=sender:from:to:cc:subject:date:message-id:in-reply-to:references
	:mime-version:content-transfer-encoding;
	bh=rXc5w1xK4r4tUT4gpCImqZbzXbplXrAcxFWe8jYu6qw=;
	b=KZUtnTi4eb/ugNZ8DLi2AMMKz11TbHx0dlCcKb00B52V9wM8IYXLljXOdHQbJRkZig
	hwcTLZuwtujuO5kXvACSx0t+Io+V4xDLXiBY783orNsx2xJK2jhkwBDEloqugmHyiosR
	xHC6GfECxP/be8cUjo4dtmjXfjO1kOyR2cE7wNRpk6VYMUzzhbt6kXYNnfnlzcijFnjj
	M62cQjQT+HkdRkvrEpiBF2Lp4aJ/Vvk7Dp8v59g/91gQaz4vrmMggNmPbltFJt4WR/Or
	JVrp7ns+Z9y43eH/jz5CDX0YGrCyFP70iA07c8LjZboAqFsUolCKDZCALD7zMHQLW/s2
	c3uw==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
	d=1e100.net; s=20161025;
	h=x-gm-message-state:sender:from:to:cc:subject:date:message-id
	:in-reply-to:references:mime-version:content-transfer-encoding;
	bh=rXc5w1xK4r4tUT4gpCImqZbzXbplXrAcxFWe8jYu6qw=;
	b=V0H+vHETjS7/IYtrrVWpgCG/zPe+v9zfT2UuaXiKMp4ciXMUINsBHHUGJnUL3R4e5B
	G/a3vcIzmTeX2h07h/vtOeCGLxbFIwrHcT3WJlrio0AOQwJ3B7cmoKY2xT6+pcamgF27
	AhmhDN+A69AovE7z6rMmy/WWGr8/yjvbSVwrtUdUfA13oxfDeWWcLzQbkmTSz0x3HHTq
	s/AWd8otuDRolaokoXivq412gJsuYRC2hPfST/bOXUYPMrYB4+S4cjDYmleChOYoSuE9
	rioLb4UFRhIePH/cQed8s9Z51HAEKFWb6vorzUxX724Ttund/uMhu0RBAuVeh3HAXfAI
	NG5Q==
X-Gm-Message-State: ABuFfojA4JYuHN/F8D22v7yWWEpWx7h7WxoqLchyDqkGRdt/d4xSiPKM
	yKmDM7BN/5I4sWQUaq2BRLk=
X-Google-Smtp-Source: 
 ACcGV63vA9wID35i8L8BPd9R6AKFZHQ4jr7e05NFIfhhi7N0K5tAkBo6sFO1NSV3GPwltVwviVrcMQ==
X-Received: by 2002:a1c:3ec3:: with SMTP id
	l186-v6mr2273299wma.84.1539211026916;
	Wed, 10 Oct 2018 15:37:06 -0700 (PDT)
Received: from x1.local (26.red-83-32-208.dynamicip.rima-tde.net.
	[83.32.208.26]) by smtp.gmail.com with ESMTPSA id
	q77-v6sm10344886wmd.33.2018.10.10.15.37.05
	(version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
	Wed, 10 Oct 2018 15:37:06 -0700 (PDT)
From: =?utf-8?q?Philippe_Mathieu-Daud=C3=A9?= <f4bug@amsat.org>
To: Laurent Vivier <laurent@vivier.eu>
Date: Thu, 11 Oct 2018 00:36:51 +0200
Message-Id: <20181010223656.31632-6-f4bug@amsat.org>
X-Mailer: git-send-email 2.19.1
In-Reply-To: <20181010223656.31632-1-f4bug@amsat.org>
References: <20181010223656.31632-1-f4bug@amsat.org>
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
X-detected-operating-system: by eggs.gnu.org: Genre and OS details not
	recognized.
X-Received-From: 2a00:1450:4864:20::332
Subject: [Qemu-devel] [PATCH v5 05/10] linux-user/syscall: Introduce
 target_sockaddr_nl
X-BeenThere: qemu-devel@nongnu.org
X-Mailman-Version: 2.1.21
Precedence: list
List-Id: <qemu-devel.nongnu.org>
List-Unsubscribe: <https://lists.nongnu.org/mailman/options/qemu-devel>,
	<mailto:qemu-devel-request@nongnu.org?subject=unsubscribe>
List-Archive: <http://lists.nongnu.org/archive/html/qemu-devel/>
List-Post: <mailto:qemu-devel@nongnu.org>
List-Help: <mailto:qemu-devel-request@nongnu.org?subject=help>
List-Subscribe: <https://lists.nongnu.org/mailman/listinfo/qemu-devel>,
	<mailto:qemu-devel-request@nongnu.org?subject=subscribe>
Cc: =?utf-8?q?Guido_G=C3=BCnther?= <agx@sigxcpu.org>,
 Riku Voipio <riku.voipio@iki.fi>,
 =?utf-8?q?Philippe_Mathieu-Daud=C3=A9?= <f4bug@amsat.org>,
 qemu-devel@nongnu.org
Errors-To: 
 qemu-devel-bounces+patchwork-qemu-devel=patchwork.kernel.org@nongnu.org
Sender: "Qemu-devel"
	<qemu-devel-bounces+patchwork-qemu-devel=patchwork.kernel.org@nongnu.org>
X-Virus-Scanned: ClamAV using ClamSMTP

Signed-off-by: Philippe Mathieu-Daudé <f4bug@amsat.org>
Tested-By: Guido Günther <agx@sigxcpu.org>
---
 linux-user/syscall_defs.h | 7 +++++++
 linux-user/syscall.c      | 6 ++++--
 2 files changed, 11 insertions(+), 2 deletions(-)

diff --git a/linux-user/syscall_defs.h b/linux-user/syscall_defs.h
index 18d434d6dc..8ddeec1bae 100644
--- a/linux-user/syscall_defs.h
+++ b/linux-user/syscall_defs.h
@@ -152,6 +152,13 @@ struct target_sockaddr_un {
     uint8_t sun_path[108];
 };
 
+struct target_sockaddr_nl {
+    uint16_t nl_family;     /* AF_NETLINK */
+    uint16_t __pad;
+    uint32_t nl_pid;
+    uint32_t nl_groups;
+};
+
 struct target_in_addr {
     uint32_t s_addr; /* big endian */
 };
diff --git a/linux-user/syscall.c b/linux-user/syscall.c
index ea503381aa..e9cadfd748 100644
--- a/linux-user/syscall.c
+++ b/linux-user/syscall.c
@@ -1447,8 +1447,10 @@ static inline abi_long host_to_target_sockaddr(abi_ulong target_addr,
         sizeof(target_saddr->sa_family)) {
         target_saddr->sa_family = tswap16(addr->sa_family);
     }
-    if (addr->sa_family == AF_NETLINK && len >= sizeof(struct sockaddr_nl)) {
-        struct sockaddr_nl *target_nl = (struct sockaddr_nl *)target_saddr;
+    if (addr->sa_family == AF_NETLINK &&
+        len >= sizeof(struct target_sockaddr_nl)) {
+        struct target_sockaddr_nl *target_nl =
+               (struct target_sockaddr_nl *)target_saddr;
         target_nl->nl_pid = tswap32(target_nl->nl_pid);
         target_nl->nl_groups = tswap32(target_nl->nl_groups);
     } else if (addr->sa_family == AF_PACKET) {

From patchwork Wed Oct 10 22:36:52 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit
X-Patchwork-Submitter: =?utf-8?q?Philippe_Mathieu-Daud=C3=A9?=
 <f4bug@amsat.org>
X-Patchwork-Id: 10635405
Return-Path: 
 <qemu-devel-bounces+patchwork-qemu-devel=patchwork.kernel.org@nongnu.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 70C6915E2
	for <patchwork-qemu-devel@patchwork.kernel.org>;
 Wed, 10 Oct 2018 22:43:21 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 4FA1C287D1
	for <patchwork-qemu-devel@patchwork.kernel.org>;
 Wed, 10 Oct 2018 22:43:21 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 3D82528842; Wed, 10 Oct 2018 22:43:21 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.7 required=2.0 tests=BAYES_00,DKIM_INVALID,
	DKIM_SIGNED,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from lists.gnu.org (lists.gnu.org [208.118.235.17])
	(using TLSv1 with cipher AES256-SHA (256/256 bits))
	(No client certificate requested)
	by mail.wl.linuxfoundation.org (Postfix) with ESMTPS id D4C7A287D1
	for <patchwork-qemu-devel@patchwork.kernel.org>;
 Wed, 10 Oct 2018 22:43:20 +0000 (UTC)
Received: from localhost ([::1]:59460 helo=lists.gnu.org)
	by lists.gnu.org with esmtp (Exim 4.71)
	(envelope-from
 <qemu-devel-bounces+patchwork-qemu-devel=patchwork.kernel.org@nongnu.org>)
	id 1gANCK-0007Aq-90
	for patchwork-qemu-devel@patchwork.kernel.org;
 Wed, 10 Oct 2018 18:43:20 -0400
Received: from eggs.gnu.org ([2001:4830:134:3::10]:38814)
	by lists.gnu.org with esmtp (Exim 4.71)
	(envelope-from <philippe.mathieu.daude@gmail.com>)
	id 1gAN6O-0002fY-6m
	for qemu-devel@nongnu.org; Wed, 10 Oct 2018 18:37:14 -0400
Received: from Debian-exim by eggs.gnu.org with spam-scanned (Exim 4.71)
	(envelope-from <philippe.mathieu.daude@gmail.com>)
	id 1gAN6N-0000QE-C5
	for qemu-devel@nongnu.org; Wed, 10 Oct 2018 18:37:12 -0400
Received: from mail-wm1-x341.google.com ([2a00:1450:4864:20::341]:32912)
	by eggs.gnu.org with esmtps (TLS1.0:RSA_AES_128_CBC_SHA1:16)
	(Exim 4.71) (envelope-from <philippe.mathieu.daude@gmail.com>)
	id 1gAN6M-0000Me-Be
	for qemu-devel@nongnu.org; Wed, 10 Oct 2018 18:37:11 -0400
Received: by mail-wm1-x341.google.com with SMTP id y140-v6so14419168wmd.0
	for <qemu-devel@nongnu.org>; Wed, 10 Oct 2018 15:37:09 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=gmail.com; s=20161025;
	h=sender:from:to:cc:subject:date:message-id:in-reply-to:references
	:mime-version:content-transfer-encoding;
	bh=cYlG1y4UEcFo2vUs6TAwVRoCFVbsQAwf5hFRy30uqhs=;
	b=Pa8t+aWezXY0nSxsphnquqllOTGn/V981CRbYTnspuBUidorbL4ptXLbkb5Hc2K/08
	RrUxjtwKSs8CggpzjjLnlpKAJMcEiWC4tcUjZd6V+Y43cXsJoJnXbhhF4OF2kG1lEYhE
	9vjClOVqrVtK6S0SheGApWeCx5WRwc6qa9HM1GdKfGZQF5zYNpUz7gbKcSPOCIkaPvZH
	y7aNgDBBGZPrm1HW04eFM1MTsMYuznaKF1UGrO7KOEVpyCUqz7RnMWNN/FbUTPvuHHx+
	tNpqcb2rZC4mv/E1sTZLHxvpEjsnTpZHbMy4DF0mfwEUcin/bRr7bjdOHWubERmRUdYT
	JM4w==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
	d=1e100.net; s=20161025;
	h=x-gm-message-state:sender:from:to:cc:subject:date:message-id
	:in-reply-to:references:mime-version:content-transfer-encoding;
	bh=cYlG1y4UEcFo2vUs6TAwVRoCFVbsQAwf5hFRy30uqhs=;
	b=lpJ7ZuRsOJDSUNaFtidBcgbwGuyB6E+decnoR8m+OJ5C1eV4+U4jFce9vlmZYPIC2O
	levDJ6/hxiYA2hcShJpcrOydsQSk6q6OCVobNHK7X99zumYqBE6zysfWXZ9ZcVTUZoN3
	Espx5/ck0/DEMcpwYkiWbCx4sNi2m1az1eLVvvY1Wr9m5iKaB6ThE1VHBPTVsm4BMx9Z
	z4v1kbdNXi1qaIcAML68H13n5V+VXYescrlgHZQKowaQDXLOPwSSJhRi+b9U+uiFHbFP
	28MJR9pLOubXfYq2r1HSi15EDFL/xw5SxNS9DszzXszBlbojf7MNatXkCptuC+2cgin3
	6yBg==
X-Gm-Message-State: ABuFfoitVLTjk9cFJxU8SnvrhFX0ZJqszbq9qw2R4mXgwG73qdVVri0C
	t8al7J8Jk6lQyxnhw+FqSDM=
X-Google-Smtp-Source: 
 ACcGV63aRsM/3+52gcxrFepQd139cdBMnM5zCjgR2nMyOGyPu1A3Iph4o6oPrd4o0FMy7TQv04JnFg==
X-Received: by 2002:a1c:bc82:: with SMTP id
	m124-v6mr2340315wmf.76.1539211028151;
	Wed, 10 Oct 2018 15:37:08 -0700 (PDT)
Received: from x1.local (26.red-83-32-208.dynamicip.rima-tde.net.
	[83.32.208.26]) by smtp.gmail.com with ESMTPSA id
	q77-v6sm10344886wmd.33.2018.10.10.15.37.07
	(version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
	Wed, 10 Oct 2018 15:37:07 -0700 (PDT)
From: =?utf-8?q?Philippe_Mathieu-Daud=C3=A9?= <f4bug@amsat.org>
To: Laurent Vivier <laurent@vivier.eu>
Date: Thu, 11 Oct 2018 00:36:52 +0200
Message-Id: <20181010223656.31632-7-f4bug@amsat.org>
X-Mailer: git-send-email 2.19.1
In-Reply-To: <20181010223656.31632-1-f4bug@amsat.org>
References: <20181010223656.31632-1-f4bug@amsat.org>
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
X-detected-operating-system: by eggs.gnu.org: Genre and OS details not
	recognized.
X-Received-From: 2a00:1450:4864:20::341
Subject: [Qemu-devel] [PATCH v5 06/10] linux-user/strace: Dump AF_NETLINK
 sockaddr content
X-BeenThere: qemu-devel@nongnu.org
X-Mailman-Version: 2.1.21
Precedence: list
List-Id: <qemu-devel.nongnu.org>
List-Unsubscribe: <https://lists.nongnu.org/mailman/options/qemu-devel>,
	<mailto:qemu-devel-request@nongnu.org?subject=unsubscribe>
List-Archive: <http://lists.nongnu.org/archive/html/qemu-devel/>
List-Post: <mailto:qemu-devel@nongnu.org>
List-Help: <mailto:qemu-devel-request@nongnu.org?subject=help>
List-Subscribe: <https://lists.nongnu.org/mailman/listinfo/qemu-devel>,
	<mailto:qemu-devel-request@nongnu.org?subject=subscribe>
Cc: =?utf-8?q?Guido_G=C3=BCnther?= <agx@sigxcpu.org>,
 Riku Voipio <riku.voipio@iki.fi>,
 =?utf-8?q?Philippe_Mathieu-Daud=C3=A9?= <f4bug@amsat.org>,
 qemu-devel@nongnu.org
Errors-To: 
 qemu-devel-bounces+patchwork-qemu-devel=patchwork.kernel.org@nongnu.org
Sender: "Qemu-devel"
	<qemu-devel-bounces+patchwork-qemu-devel=patchwork.kernel.org@nongnu.org>
X-Virus-Scanned: ClamAV using ClamSMTP

Signed-off-by: Philippe Mathieu-Daudé <f4bug@amsat.org>
Tested-By: Guido Günther <agx@sigxcpu.org>
---
 linux-user/strace.c | 37 +++++++++++++++++++++++++++++++++++++
 1 file changed, 37 insertions(+)

diff --git a/linux-user/strace.c b/linux-user/strace.c
index ca2cd75993..425c06d169 100644
--- a/linux-user/strace.c
+++ b/linux-user/strace.c
@@ -8,6 +8,7 @@
 #include <arpa/inet.h>
 #include <netinet/tcp.h>
 #include <linux/if_packet.h>
+#include <linux/netlink.h>
 #include <sched.h>
 #include "qemu.h"
 
@@ -398,6 +399,12 @@ print_sockaddr(abi_ulong addr, abi_long addrlen)
             gemu_log("}");
             break;
         }
+        case AF_NETLINK: {
+            struct target_sockaddr_nl *nl = (struct target_sockaddr_nl *)sa;
+            gemu_log("{nl_family=AF_NETLINK,nl_pid=%u,nl_groups=%u}",
+                     nl->nl_pid, nl->nl_groups);
+            break;
+        }
         default:
             gemu_log("{sa_family=%d, sa_data={", sa->sa_family);
             for (i = 0; i < 13; i++) {
@@ -424,6 +431,9 @@ print_socket_domain(int domain)
     case PF_INET:
         gemu_log("PF_INET");
         break;
+    case PF_NETLINK:
+        gemu_log("PF_NETLINK");
+        break;
     case PF_PACKET:
         gemu_log("PF_PACKET");
         break;
@@ -473,6 +483,33 @@ print_socket_protocol(int domain, int type, int protocol)
         return;
     }
 
+    if (domain == PF_NETLINK) {
+        switch (protocol) {
+        case NETLINK_ROUTE:
+            gemu_log("NETLINK_ROUTE");
+            break;
+        case NETLINK_AUDIT:
+            gemu_log("NETLINK_AUDIT");
+            break;
+        case NETLINK_NETFILTER:
+            gemu_log("NETLINK_NETFILTER");
+            break;
+        case NETLINK_KOBJECT_UEVENT:
+            gemu_log("NETLINK_KOBJECT_UEVENT");
+            break;
+        case NETLINK_RDMA:
+            gemu_log("NETLINK_RDMA");
+            break;
+        case NETLINK_CRYPTO:
+            gemu_log("NETLINK_CRYPTO");
+            break;
+        default:
+            gemu_log("%d", protocol);
+            break;
+        }
+        return;
+    }
+
     switch (protocol) {
     case IPPROTO_IP:
         gemu_log("IPPROTO_IP");

From patchwork Wed Oct 10 22:36:53 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit
X-Patchwork-Submitter: =?utf-8?q?Philippe_Mathieu-Daud=C3=A9?=
 <f4bug@amsat.org>
X-Patchwork-Id: 10635397
Return-Path: 
 <qemu-devel-bounces+patchwork-qemu-devel=patchwork.kernel.org@nongnu.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 62AEC16B1
	for <patchwork-qemu-devel@patchwork.kernel.org>;
 Wed, 10 Oct 2018 22:38:51 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 4DA162AC4A
	for <patchwork-qemu-devel@patchwork.kernel.org>;
 Wed, 10 Oct 2018 22:38:51 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 41BB52AC5D; Wed, 10 Oct 2018 22:38:51 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.7 required=2.0 tests=BAYES_00,DKIM_INVALID,
	DKIM_SIGNED,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from lists.gnu.org (lists.gnu.org [208.118.235.17])
	(using TLSv1 with cipher AES256-SHA (256/256 bits))
	(No client certificate requested)
	by mail.wl.linuxfoundation.org (Postfix) with ESMTPS id D91B92AC4A
	for <patchwork-qemu-devel@patchwork.kernel.org>;
 Wed, 10 Oct 2018 22:38:50 +0000 (UTC)
Received: from localhost ([::1]:59436 helo=lists.gnu.org)
	by lists.gnu.org with esmtp (Exim 4.71)
	(envelope-from
 <qemu-devel-bounces+patchwork-qemu-devel=patchwork.kernel.org@nongnu.org>)
	id 1gAN7y-0003tP-7M
	for patchwork-qemu-devel@patchwork.kernel.org;
 Wed, 10 Oct 2018 18:38:50 -0400
Received: from eggs.gnu.org ([2001:4830:134:3::10]:38831)
	by lists.gnu.org with esmtp (Exim 4.71)
	(envelope-from <philippe.mathieu.daude@gmail.com>)
	id 1gAN6O-0002fv-O3
	for qemu-devel@nongnu.org; Wed, 10 Oct 2018 18:37:15 -0400
Received: from Debian-exim by eggs.gnu.org with spam-scanned (Exim 4.71)
	(envelope-from <philippe.mathieu.daude@gmail.com>)
	id 1gAN6N-0000Qy-LH
	for qemu-devel@nongnu.org; Wed, 10 Oct 2018 18:37:12 -0400
Received: from mail-wr1-x42e.google.com ([2a00:1450:4864:20::42e]:39675)
	by eggs.gnu.org with esmtps (TLS1.0:RSA_AES_128_CBC_SHA1:16)
	(Exim 4.71) (envelope-from <philippe.mathieu.daude@gmail.com>)
	id 1gAN6N-0000OZ-DK
	for qemu-devel@nongnu.org; Wed, 10 Oct 2018 18:37:11 -0400
Received: by mail-wr1-x42e.google.com with SMTP id 61-v6so7451271wrb.6
	for <qemu-devel@nongnu.org>; Wed, 10 Oct 2018 15:37:10 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=gmail.com; s=20161025;
	h=sender:from:to:cc:subject:date:message-id:in-reply-to:references
	:mime-version:content-transfer-encoding;
	bh=GwUCsIlCX+6WpZBneg9X1OhKU/JQO9jslVgj3i81wac=;
	b=l55sQ4yYojSmkIXK3KpYeK0Pmnf/EfxIl7hXguPTyoDoxuyrGB7uaYlbLdueM5KyQd
	tkmCkOCvzJHGu2xQokCjr2sjq51dbcuUGFACZRp35xnHHbiL1DqBP70FlOiDcZs4K7Yv
	nr9+Hgr4SQAtdNhuWWMlG5vn0STgYItop9aCDS2wLCaPyIOJD5+dyeh+l5yi4fiDU9TE
	1jHbmGyfS4iLxSv3VoEqX3nruGjm/Pk4EhE2OUKbK+hDS1QsVx6Ddzxtq+25fivjXWDm
	Qxpb/nGpwqfqDueTok2I6QeTFlqlIMVcdAP6zfY1jRnQFFe6/l+DPBiUN+/OZdKrHlMk
	3fUQ==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
	d=1e100.net; s=20161025;
	h=x-gm-message-state:sender:from:to:cc:subject:date:message-id
	:in-reply-to:references:mime-version:content-transfer-encoding;
	bh=GwUCsIlCX+6WpZBneg9X1OhKU/JQO9jslVgj3i81wac=;
	b=Q9cZV74hEXuwP/CEnrzjHJQ40UprRpsMCRLJTo46wqMWDlJTRKCmN/kj/uOg1Fqkn4
	JXFZFh9Y1aLB7BLQwAH4CNQwgcf2epKwIleyYNFmiqvUSXBxJiTdG9sHE/STyyAPK2it
	jXk2N98IvugQ3BqeJjdC52gkBDjwo5EMkgeOIX2PqRQaawR7Lk0J9O8DVj79w6el74Kb
	9sdUSmROyTyhvuOsTSWbGvqBH96tHnQ2YbLjaTqkV+dbq4B5XjGJVPDZHq02TC0fcuT0
	kYK9ZsvME83bwbbFvGgWNZgV1QUKObhTqiuTS78OPsKrnpxJqA8K29vw1tG3W5+uIcYb
	GQdw==
X-Gm-Message-State: ABuFfoiVxEy/yKDz/gcSPpS+c/032CdR2i8JmAV0H+eiblAVxjFUbBqN
	BIFWPDG/xUgEN/Wo7pxOsEI=
X-Google-Smtp-Source: 
 ACcGV60jcSPPYR+H2u1999Ol3hrkqfreCVYW479VLSgKk2ZGmqZoDawZmpznabr09feZCIAAb/xVew==
X-Received: by 2002:adf:9b12:: with SMTP id
	b18-v6mr19413704wrc.35.1539211029380;
	Wed, 10 Oct 2018 15:37:09 -0700 (PDT)
Received: from x1.local (26.red-83-32-208.dynamicip.rima-tde.net.
	[83.32.208.26]) by smtp.gmail.com with ESMTPSA id
	q77-v6sm10344886wmd.33.2018.10.10.15.37.08
	(version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
	Wed, 10 Oct 2018 15:37:08 -0700 (PDT)
From: =?utf-8?q?Philippe_Mathieu-Daud=C3=A9?= <f4bug@amsat.org>
To: Laurent Vivier <laurent@vivier.eu>
Date: Thu, 11 Oct 2018 00:36:53 +0200
Message-Id: <20181010223656.31632-8-f4bug@amsat.org>
X-Mailer: git-send-email 2.19.1
In-Reply-To: <20181010223656.31632-1-f4bug@amsat.org>
References: <20181010223656.31632-1-f4bug@amsat.org>
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
X-detected-operating-system: by eggs.gnu.org: Genre and OS details not
	recognized.
X-Received-From: 2a00:1450:4864:20::42e
Subject: [Qemu-devel] [PATCH v5 07/10] linux-user/strace: Add print_sockfd()
X-BeenThere: qemu-devel@nongnu.org
X-Mailman-Version: 2.1.21
Precedence: list
List-Id: <qemu-devel.nongnu.org>
List-Unsubscribe: <https://lists.nongnu.org/mailman/options/qemu-devel>,
	<mailto:qemu-devel-request@nongnu.org?subject=unsubscribe>
List-Archive: <http://lists.nongnu.org/archive/html/qemu-devel/>
List-Post: <mailto:qemu-devel@nongnu.org>
List-Help: <mailto:qemu-devel-request@nongnu.org?subject=help>
List-Subscribe: <https://lists.nongnu.org/mailman/listinfo/qemu-devel>,
	<mailto:qemu-devel-request@nongnu.org?subject=subscribe>
Cc: =?utf-8?q?Guido_G=C3=BCnther?= <agx@sigxcpu.org>,
 Riku Voipio <riku.voipio@iki.fi>,
 =?utf-8?q?Philippe_Mathieu-Daud=C3=A9?= <f4bug@amsat.org>,
 qemu-devel@nongnu.org
Errors-To: 
 qemu-devel-bounces+patchwork-qemu-devel=patchwork.kernel.org@nongnu.org
Sender: "Qemu-devel"
	<qemu-devel-bounces+patchwork-qemu-devel=patchwork.kernel.org@nongnu.org>
X-Virus-Scanned: ClamAV using ClamSMTP

Extract common print_sockfd() from various socket related syscalls.

Signed-off-by: Philippe Mathieu-Daudé <f4bug@amsat.org>
---
 linux-user/strace.c | 19 ++++++++++++-------
 1 file changed, 12 insertions(+), 7 deletions(-)

diff --git a/linux-user/strace.c b/linux-user/strace.c
index 425c06d169..5435d01569 100644
--- a/linux-user/strace.c
+++ b/linux-user/strace.c
@@ -1651,6 +1651,11 @@ static void do_print_socket(const char *name, abi_long arg1)
     gemu_log(")");
 }
 
+static void print_sockfd(abi_long sockfd, int last)
+{
+    print_raw_param(TARGET_ABI_FMT_ld, sockfd, last);
+}
+
 static void do_print_sockaddr(const char *name, abi_long arg1)
 {
     abi_ulong sockfd, addr, addrlen;
@@ -1660,7 +1665,7 @@ static void do_print_sockaddr(const char *name, abi_long arg1)
     get_user_ualx(addrlen, arg1, 2);
 
     gemu_log("%s(", name);
-    print_raw_param(TARGET_ABI_FMT_ld, sockfd, 0);
+    print_sockfd(sockfd, 0);
     print_sockaddr(addr, addrlen);
     gemu_log(")");
 }
@@ -1673,7 +1678,7 @@ static void do_print_listen(const char *name, abi_long arg1)
     get_user_ualx(backlog, arg1, 1);
 
     gemu_log("%s(", name);
-    print_raw_param(TARGET_ABI_FMT_ld, sockfd, 0);
+    print_sockfd(sockfd, 0);
     print_raw_param(TARGET_ABI_FMT_ld, backlog, 1);
     gemu_log(")");
 }
@@ -1708,7 +1713,7 @@ static void do_print_sendrecv(const char *name, abi_long arg1)
     get_user_ualx(flags, arg1, 3);
 
     gemu_log("%s(", name);
-    print_raw_param(TARGET_ABI_FMT_ld, sockfd, 0);
+    print_sockfd(sockfd, 0);
     print_buf(msg, len, 0);
     print_raw_param(TARGET_ABI_FMT_ld, len, 0);
     print_flags(msg_flags, flags, 1);
@@ -1727,7 +1732,7 @@ static void do_print_msgaddr(const char *name, abi_long arg1)
     get_user_ualx(addrlen, arg1, 5);
 
     gemu_log("%s(", name);
-    print_raw_param(TARGET_ABI_FMT_ld, sockfd, 0);
+    print_sockfd(sockfd, 0);
     print_buf(msg, len, 0);
     print_raw_param(TARGET_ABI_FMT_ld, len, 0);
     print_flags(msg_flags, flags, 0);
@@ -1743,7 +1748,7 @@ static void do_print_shutdown(const char *name, abi_long arg1)
     get_user_ualx(how, arg1, 1);
 
     gemu_log("shutdown(");
-    print_raw_param(TARGET_ABI_FMT_ld, sockfd, 0);
+    print_sockfd(sockfd, 0);
     switch (how) {
     case SHUT_RD:
         gemu_log("SHUT_RD");
@@ -1770,7 +1775,7 @@ static void do_print_msg(const char *name, abi_long arg1)
     get_user_ualx(flags, arg1, 2);
 
     gemu_log("%s(", name);
-    print_raw_param(TARGET_ABI_FMT_ld, sockfd, 0);
+    print_sockfd(sockfd, 0);
     print_pointer(msg, 0);
     print_flags(msg_flags, flags, 1);
     gemu_log(")");
@@ -1787,7 +1792,7 @@ static void do_print_sockopt(const char *name, abi_long arg1)
     get_user_ualx(optlen, arg1, 4);
 
     gemu_log("%s(", name);
-    print_raw_param(TARGET_ABI_FMT_ld, sockfd, 0);
+    print_sockfd(sockfd, 0);
     switch (level) {
     case SOL_TCP:
         gemu_log("SOL_TCP,");

From patchwork Wed Oct 10 22:36:54 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit
X-Patchwork-Submitter: =?utf-8?q?Philippe_Mathieu-Daud=C3=A9?=
 <f4bug@amsat.org>
X-Patchwork-Id: 10635407
Return-Path: 
 <qemu-devel-bounces+patchwork-qemu-devel=patchwork.kernel.org@nongnu.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 7215614DB
	for <patchwork-qemu-devel@patchwork.kernel.org>;
 Wed, 10 Oct 2018 22:44:34 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 58A2B2AC4A
	for <patchwork-qemu-devel@patchwork.kernel.org>;
 Wed, 10 Oct 2018 22:44:34 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 4C91C2AC5D; Wed, 10 Oct 2018 22:44:34 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.7 required=2.0 tests=BAYES_00,DKIM_INVALID,
	DKIM_SIGNED,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from lists.gnu.org (lists.gnu.org [208.118.235.17])
	(using TLSv1 with cipher AES256-SHA (256/256 bits))
	(No client certificate requested)
	by mail.wl.linuxfoundation.org (Postfix) with ESMTPS id E903F2AC4A
	for <patchwork-qemu-devel@patchwork.kernel.org>;
 Wed, 10 Oct 2018 22:44:33 +0000 (UTC)
Received: from localhost ([::1]:59468 helo=lists.gnu.org)
	by lists.gnu.org with esmtp (Exim 4.71)
	(envelope-from
 <qemu-devel-bounces+patchwork-qemu-devel=patchwork.kernel.org@nongnu.org>)
	id 1gANDU-0007nS-Rg
	for patchwork-qemu-devel@patchwork.kernel.org;
 Wed, 10 Oct 2018 18:44:32 -0400
Received: from eggs.gnu.org ([2001:4830:134:3::10]:38838)
	by lists.gnu.org with esmtp (Exim 4.71)
	(envelope-from <philippe.mathieu.daude@gmail.com>)
	id 1gAN6R-0002g1-2r
	for qemu-devel@nongnu.org; Wed, 10 Oct 2018 18:37:18 -0400
Received: from Debian-exim by eggs.gnu.org with spam-scanned (Exim 4.71)
	(envelope-from <philippe.mathieu.daude@gmail.com>)
	id 1gAN6N-0000RK-Vk
	for qemu-devel@nongnu.org; Wed, 10 Oct 2018 18:37:12 -0400
Received: from mail-wm1-x32a.google.com ([2a00:1450:4864:20::32a]:50248)
	by eggs.gnu.org with esmtps (TLS1.0:RSA_AES_128_CBC_SHA1:16)
	(Exim 4.71) (envelope-from <philippe.mathieu.daude@gmail.com>)
	id 1gAN6N-0000Or-Ms
	for qemu-devel@nongnu.org; Wed, 10 Oct 2018 18:37:11 -0400
Received: by mail-wm1-x32a.google.com with SMTP id i8-v6so7017958wmg.0
	for <qemu-devel@nongnu.org>; Wed, 10 Oct 2018 15:37:11 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=gmail.com; s=20161025;
	h=sender:from:to:cc:subject:date:message-id:in-reply-to:references
	:mime-version:content-transfer-encoding;
	bh=9UGvxbUbOUkUs2h21ghA50gksaYfYt0H93Hwt6BIzRw=;
	b=HH6XFs47KWxzSGy26l5F+zcRAJt7pwgyW30vQwbnIEyUNpjKAobaW7gmro/7wjL/wR
	wTnF92aaEnWGSKmieSQuhH4tcqCnBws+xxC9GoU1hJgQ5Orv7Bb89pUE3ww4fXAryIcu
	VuYvsgn2WJ4oGw1afKUCMaXcRunJDxdtFcZqVpqF/WoKBw8tmQC7tCQ3kqibZMIs3TxO
	kc/ybVSaKbuOrhsgISpSrPAGGtRSaf5+qjrGkI/K3/IFNJ7OjuKpJG7MeqqmfQVtLEat
	5Sems45glw43xD6Xi/9vcuAMi41ThbCpovdrxFeumdYFUlc/hJ6RdxrE+uQfboDin1aJ
	gEyg==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
	d=1e100.net; s=20161025;
	h=x-gm-message-state:sender:from:to:cc:subject:date:message-id
	:in-reply-to:references:mime-version:content-transfer-encoding;
	bh=9UGvxbUbOUkUs2h21ghA50gksaYfYt0H93Hwt6BIzRw=;
	b=tG/IW5mF4MUQVFAabjWGPgyD3LNxgfUHmrPhweY8NpGQmaeAxR0/9Bciwwq0bhTbWE
	RmqyKY2yWRSWgaSo3gMJyvboQAAmP2A7kO6OUyZ9YMMgnU74/aYgF8NOax37kTZpqdJw
	INqJdQoJs2FeJyo85f/wyRKDy6tURvj3UuHe6c4zxYL2ZHX7N+2TRD1uic/anpnQLuoV
	hyU6klySpxR2VIfo7rLirtow1Vb7UhIY4jpjr6lkpXjY1A1Fo1VHY3pJhKWF9fKM0AR2
	Prw734hajqRsdKz/nkY88Zbiep9pR4P3Vv8qykBwLX8/NQ5RVKVV6RxTYlNI3h+tKzMj
	+Omw==
X-Gm-Message-State: ABuFfojyRudrZCC+G97zHy8Xc/nJeaiUlPMihhhgG3qTtH1W+GQT/ckK
	9pKe+BKVm32K0JFjEikaAUw=
X-Google-Smtp-Source: 
 ACcGV61IaXszHIwVKStBnGVmDj88apEmkudj1w4yi03eQC6Lkbz5U3a9Ny/YJtsUHTnb021kKgaD7w==
X-Received: by 2002:a1c:6655:: with SMTP id
	a82-v6mr2505530wmc.10.1539211030741;
	Wed, 10 Oct 2018 15:37:10 -0700 (PDT)
Received: from x1.local (26.red-83-32-208.dynamicip.rima-tde.net.
	[83.32.208.26]) by smtp.gmail.com with ESMTPSA id
	q77-v6sm10344886wmd.33.2018.10.10.15.37.09
	(version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
	Wed, 10 Oct 2018 15:37:10 -0700 (PDT)
From: =?utf-8?q?Philippe_Mathieu-Daud=C3=A9?= <f4bug@amsat.org>
To: Laurent Vivier <laurent@vivier.eu>
Date: Thu, 11 Oct 2018 00:36:54 +0200
Message-Id: <20181010223656.31632-9-f4bug@amsat.org>
X-Mailer: git-send-email 2.19.1
In-Reply-To: <20181010223656.31632-1-f4bug@amsat.org>
References: <20181010223656.31632-1-f4bug@amsat.org>
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
X-detected-operating-system: by eggs.gnu.org: Genre and OS details not
	recognized.
X-Received-From: 2a00:1450:4864:20::32a
Subject: [Qemu-devel] [PATCH v5 08/10] linux-user/strace: Let
 print_sockaddr() have a 'last' argument
X-BeenThere: qemu-devel@nongnu.org
X-Mailman-Version: 2.1.21
Precedence: list
List-Id: <qemu-devel.nongnu.org>
List-Unsubscribe: <https://lists.nongnu.org/mailman/options/qemu-devel>,
	<mailto:qemu-devel-request@nongnu.org?subject=unsubscribe>
List-Archive: <http://lists.nongnu.org/archive/html/qemu-devel/>
List-Post: <mailto:qemu-devel@nongnu.org>
List-Help: <mailto:qemu-devel-request@nongnu.org?subject=help>
List-Subscribe: <https://lists.nongnu.org/mailman/listinfo/qemu-devel>,
	<mailto:qemu-devel-request@nongnu.org?subject=subscribe>
Cc: =?utf-8?q?Guido_G=C3=BCnther?= <agx@sigxcpu.org>,
 Riku Voipio <riku.voipio@iki.fi>,
 =?utf-8?q?Philippe_Mathieu-Daud=C3=A9?= <f4bug@amsat.org>,
 qemu-devel@nongnu.org
Errors-To: 
 qemu-devel-bounces+patchwork-qemu-devel=patchwork.kernel.org@nongnu.org
Sender: "Qemu-devel"
	<qemu-devel-bounces+patchwork-qemu-devel=patchwork.kernel.org@nongnu.org>
X-Virus-Scanned: ClamAV using ClamSMTP

If the format is not the syscall last argument, a comma is append.

Signed-off-by: Philippe Mathieu-Daudé <f4bug@amsat.org>
Reviewed-by: Laurent Vivier <laurent@vivier.eu>
---
 linux-user/strace.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/linux-user/strace.c b/linux-user/strace.c
index 5435d01569..14ec5f9f23 100644
--- a/linux-user/strace.c
+++ b/linux-user/strace.c
@@ -67,7 +67,7 @@ UNUSED static void print_timeval(abi_ulong, int);
 UNUSED static void print_timezone(abi_ulong, int);
 UNUSED static void print_number(abi_long, int);
 UNUSED static void print_signal(abi_ulong, int);
-UNUSED static void print_sockaddr(abi_ulong addr, abi_long addrlen);
+UNUSED static void print_sockaddr(abi_ulong addr, abi_long addrlen, int);
 UNUSED static void print_socket_domain(int domain);
 UNUSED static void print_socket_type(int type);
 UNUSED static void print_socket_protocol(int domain, int type, int protocol);
@@ -336,7 +336,7 @@ static void print_siginfo(const target_siginfo_t *tinfo)
 }
 
 static void
-print_sockaddr(abi_ulong addr, abi_long addrlen)
+print_sockaddr(abi_ulong addr, abi_long addrlen, int last)
 {
     struct target_sockaddr *sa;
     int i;
@@ -418,7 +418,7 @@ print_sockaddr(abi_ulong addr, abi_long addrlen)
     } else {
         print_raw_param("0x"TARGET_ABI_FMT_lx, addr, 0);
     }
-    gemu_log(", "TARGET_ABI_FMT_ld, addrlen);
+    gemu_log(", "TARGET_ABI_FMT_ld"%s", addrlen, get_comma(last));
 }
 
 static void
@@ -1666,7 +1666,7 @@ static void do_print_sockaddr(const char *name, abi_long arg1)
 
     gemu_log("%s(", name);
     print_sockfd(sockfd, 0);
-    print_sockaddr(addr, addrlen);
+    print_sockaddr(addr, addrlen, 0);
     gemu_log(")");
 }
 
@@ -1736,7 +1736,7 @@ static void do_print_msgaddr(const char *name, abi_long arg1)
     print_buf(msg, len, 0);
     print_raw_param(TARGET_ABI_FMT_ld, len, 0);
     print_flags(msg_flags, flags, 0);
-    print_sockaddr(addr, addrlen);
+    print_sockaddr(addr, addrlen, 0);
     gemu_log(")");
 }
 
